src/scripts/booking.js

```diff
@@ -0,0 +1,191 @@
+      </div>
+    </div>
+  </div>
+  `;
+  
+  body.insertAdjacentHTML("afterbegin", firstTemplate);
+  body.style.overflow = "hidden";
+
+  // Submit
+  const submitBtn = document.querySelector("#submitBtnOne");
+  submitBtn.addEventListener('click', () => {
+    submitStepOne(challenge);
+  });
+
+  addCancelOption();
+}
+
+// Submit Step 1
+async function submitStepOne(challenge) {
+  date = document.querySelector("#bookingDate").value;
+
+  if (!date) {
+    alert('Please fill the date field!');
+    return;
+  }
+
+  const res = await fetch(`${api}/booking/available-times?date=${date}&challenge=${challenge}`);
+  times = await res.json();
+  times = times.slots;
+
+  if(!times) {
+    alert('That date is not available, try another one!')
+    return;
+  }
+
+  bookStepTwo();
+}
+
+// Book a room (Second Step)
+function bookStepTwo() {
+  const bookingPartOneDiv = document.querySelector('#bookingPartOne');
+  body.removeChild(bookingPartOneDiv);
+
+  const secondTemplate = `
+  <div class="booking" id="bookingPartTwo">
+    <div class="second-modal">
+        <h2>Book room "${challengeTitle}" (Step 2)</h2>
+
+        <label for="name">Name</label>
+        <input type="text" name="name" id="name">
+
+        <label for="email">E-mail</label>
+        <input type="email" name="email" id="email">
+
+        <label for="time">What time?</label>
+        <select name="time" id="time"></select>
+
+        <label for="participants">How many participants?</label>
+        <select name="participants" id="participants">
+            <option value="2">2 participants</option>
+            <option value="3">3 participants</option>
+            <option value="4">4 participants</option>
+            <option value="5">5 participants</option>
+            <option value="6">6 participants</option>
+        </select>
+
+        <div class="btn-parent">
+        <button id="submitBtnTwo" class="button primary">Submit booking</button>
+    </div>
+    </div>
+  </div>
+  `;
+ 
+  body.insertAdjacentHTML("afterbegin", secondTemplate);
+
+  let dateTemplate;
+  times.forEach((time)=> {
+    dateTemplate += `<option value="${time}">${time}</option>`
+  });
+
+  const time = document.querySelector("#time");
+  time.insertAdjacentHTML("afterbegin", dateTemplate);
+
+  // Submit
+  const submitBtnTwo = document.querySelector("#submitBtnTwo");
+  submitBtnTwo.addEventListener('click', () => {
+    const name = document.querySelector('#name').value;
+    const email = document.querySelector('#email').value;
+
+    // Time selected
+    const timeDiv = document.querySelector('#time');
+    const timeSelected = timeDiv.options[timeDiv.selectedIndex].text;
+
+    // Participants
+    let particiDiv = document.querySelector('#participants');
+    let participantSelected = particiDiv.options[particiDiv.selectedIndex].value;
+    participantSelected = parseInt(participantSelected);
+
+    submitStepTwo(challengeNumber, name,  email, date, timeSelected, participantSelected);
```

Comment:
To me, this code was confusing until I realized that `submitStepTwo()` doesn't just submit step two, but also proceeds to step 3. I would have moved the code to proceed from step two to step three into a separate function, or just to after this line (i.e. line 139). That way, a reader of the code can see that two things happen here: First the data from step two is submitted, and then the user is presented with step three.